### PR TITLE
Simple Replicator

### PR DESCRIPTION
**Pull request by seancookr**
This provides a simple replicator to do DB to DB replication.

This currently supports only https://wiki.apache.org/couchdb/Replication#One-shot_Replication

### CODE REVIEW

**Comment by javierg** [comment]
👏 

A nice interface will be to have instances of a Database object, where we can do something like

`database_foo.replicate database_bar`

File: lib/dolly/request.rb [hunk 1]
```diff
@@ -80,6 +80,10 @@ def request method, resource, data = nil
       end
     end
 
+    def replicate! target_db, opts={}
+      Dolly::SimpleReplicator.new(self, target_db, opts).replicate!
```
Comment (javierg) [inline lib/dolly/request.rb]: 🎉

**Comment by javierg** [comment]
cool... waiting for tests for ship it

File: lib/dolly/request.rb (hunk 1 above)
Comment (seancookr) [reply lib/dolly/request.rb]: This cant work now that I think about it. simple replicator requires request which must be required by this in order to call it as a method.  I think I will have to make another separate object for this. `Dolly::Replication.new` seems appropriate.

Comment (seancookr) [reply lib/dolly/request.rb]: this can have a `run!` method.

**Pull request closed without merging**